test/test_torch.py

```diff
@@ -3456,6 +3456,23 @@ def _test_in_place_broadcastable(t0, t1, t2=None):
                 _test_in_place_broadcastable(small2, small_expanded, large_expanded)
                 _test_in_place_broadcastable(small2, small, large)
 
+    def test_index_put_nondeterministic_alert(self, device):
+        def test_func(slf, accumulate):
+            S = 10
+            a = torch.randn(S, device=device)
+            a.index_put((torch.tensor(0, device=device),), torch.zeros((), device=device), accumulate=True)
+
+        @expectedAlertNondeterministic('index_put CPU with `accumulate=True`', 'cpu')
+        def test_func_expect_error(slf, accumulate):
+            test_func(slf, accumulate)
+
+        if torch.device(device).type == 'cpu':
+            test_func_expect_error(self, True)
+        else:
+            test_func(self, True)
```

Comment:
Does the CUDA part of this test end up doing anything? Maybe we should decorate the test with @onlyCPU?

Should we create a test_determinism.py test suite?

Reply:
Yeah, a `test_determinism.py` suite is probably a good idea. I'll open an issue.